cpp/Printf.cpp

```diff
@@ -0,0 +1,46 @@
+#include "Printf.hpp"
+
+#include <cstdio>
+#include <cstring>
+
+using namespace hsd;
+Result<void, runtime_error> hsd::xvprintf(const char* fmt, xtuple_iterator tbegin, xtuple_iterator tend) noexcept {
+    while (*fmt) {
+        char ch = *fmt++;
+        if (ch != '%')
+            std::putchar(ch);
+        else {
+            char fch = *fmt++;
+            if (!fch) {
+                return runtime_error("\nError: Invalid format string");
+            }
+            if (tbegin == tend) {
+                return runtime_error("\nError: Passed to few arguments");
+            }
+            switch (fch) {
+                case 'd': {
+                    int value = tbegin.getnext<int>();
+                    std::printf("%d", value);
```

Comment:
Maybe try to convert it to a string, because if you are trying to write a `printf` replacer you should not use `printf`

Reply:
Good idea